lib/honeybadger/backend/server.rb

```diff
@@ -48,8 +49,103 @@ def check_in(id)
+      #
+      # @param [String] project_id The unique project id
+      # @returns [Array<CheckIn>] All checkins for this project
+      # @raises CheckInSyncError on error
+      def get_check_ins(project_id)
+        response = Response.new(@http.get("/v2/projects/#{project_id}/check_ins", personal_auth_headers))
+        if response.success?
+          all_check_ins = JSON.parse(response.body)["results"]
+          return all_check_ins.map{|cfg| CheckIn.from_remote(project_id, cfg) }
+        end
+        raise CheckInSyncError.new "Fetching CheckIns failed (Code: #{response.code}) #{response.body}"
+      end
+
+      # Create check_in on project
+      # @example
+      #   backend.create_check_in('1234', check_in)
+      #
+      # @param [String] project_id The unique project id
+      # @param [CheckIn] check_in_config A CheckIn object encapsulating the config
+      # @returns [CheckIn] A CheckIn object additionally containing the id
+      # @raises CheckInSyncError on error
+      def create_check_in(project_id, check_in_config)
+        response = Response.new(@http.post("/v2/projects/#{project_id}/check_ins", check_in_config.to_json, personal_auth_headers))
+        if response.success?
+          return CheckIn.from_remote(project_id, JSON.parse(response.body))
+        end
+        raise CheckInSyncError.new "Creating CheckIn failed (Code: #{response.code}) #{response.body}"
+      end
+
+      # Update check_in on project
+      # @example
+      #   backend.update_check_in('1234', 'eajaj', check_in)
+      #
+      # @param [String] project_id The unique project id
+      # @param [String] id The unique check_in id
+      # @param [CheckIn] check_in_config A CheckIn object encapsulating the config
+      # @returns [CheckIn] updated CheckIn object
+      # @raises CheckInSyncError on error
+      def update_check_in(project_id, id, check_in_config)
+        response = Response.new(@http.put("/v2/projects/#{project_id}/check_ins/#{id}", check_in_config.to_json, personal_auth_headers))
+        if response.success?
+          return check_in_config
+        end
+        raise CheckInSyncError.new "Updating CheckIn failed (Code: #{response.code}) #{response.body}"
+      end
+
+      # Delete check_in
+      # @example
+      #   backend.delete_check_in('1234', 'eajaj')
+      #
+      # @param [String] project_id The unique project id
+      # @param [String] id The unique check_in id
+      # @returns [Boolean] true if deletion was successful
+      # @raises CheckInSyncError on error
+      def delete_check_in(project_id, id)
+        response = Response.new(@http.delete("/v2/projects/#{project_id}/check_ins/#{id}", personal_auth_headers))
+        if response.success?
+          return true
+        end
+        raise CheckInSyncError.new "Deleting CheckIn failed (Code: #{response.code}) #{response.body}"
+      end
+
       private
 
+      def personal_auth_headers
+        {"Authorization" => "#{@personal_auth_token}:"}
```

Comment:
Is this how HTTP Basic Auth is done?

Reply:
Doesn't this need to be base64 encoded?

Reply:
Yeah, no, this looks dubious on first glance. I'll test and correct.

Reply:
Ooof, I completely mocked myself here. As in: I never really tested this against the real thing and had no good tests for the authentication. Sorry for that. I also realised that I needed to refactor the server access a bit, as the new API sits on `app.honeybadger.io` instead of `api.honeybadger.io`.  I have now tested this successfully and cleaned up the code a bit as well.

@subzero10 Please take another look.